Apps/W1/HybridGP/app/src/Migration/Support/HelperFunctions.codeunit.al

```diff
@@ -2005,4 +2005,21 @@ Codeunit 4037 "Helper Functions"
                 GLAccount.Modify(true);
             end;
     end;
+
+    procedure CreateCurrencyIfNeeded(CurrencyCode: Code[10])
+    var
+        Currency: Record Currency;
+        GPMC40200: Record "GP MC40200";
+    begin
+        if (CurrencyCode <> '') and not Currency.Get(CurrencyCode) then begin
```

Comment:
Split the if condition, there should be a check if the currency code is blank and then if the currency exists. We will execute a get call on the blank currency since AL runs all conditions.

Reply:
You can also store the existing currencies in a dictionary so we avoid a SQL call, the procedure could be faster.